### PR TITLE
CORE-8860: Add TLS type to Mgm registration context

### PR DESCRIPTION
**Pull request by yift-r3**
Adds `corda.group.tls.type` to Mgm registration context

The runtime pull request is in https://github.com/corda/corda-runtime-os/pull/2890

### CODE REVIEW

**Review by charlieR3 — DISMISSED** [review]

File: data/membership-schema/src/main/resources/net/corda/schema/membership/mgm/registration/1.0/corda.mgm.registration.json [hunk 1]
```diff
@@ -79,6 +79,14 @@
         "1.3"
       ]
     },
+    "corda.group.tls.type": {
+      "description": "The TLS type. Default to OneWay",
```
Comment (williamvigorr3) [inline data/membership-schema/src/main/resources/net/corda/schema/membership/mgm/registration/1.0/corda.mgm.registration.json]: We could add more detail here. e.g. copy the docs from the gateway json schema.
```
          "description": "What kind of TLS connections to establish between the gateways in our cluster and gateways in other clusters. This setting MUST be the same on all Gateways that need to communicate via HTTPS.",
          "oneOf": [
            {"const": "ONE_WAY", "title": "Establishes a regular TLS connection, where the server gateway will be authenticated."},
            {"const": "MUTUAL", "title": "Establishes a mutual TLS connection, where both the server and client gateway are authenticated."}
```

Comment (yift-r3) [reply data/membership-schema/src/main/resources/net/corda/schema/membership/mgm/registration/1.0/corda.mgm.registration.json]: I want to make it case insensitive, so I prefer not to put `oneOf` (as the rest of the doc).

Comment (williamvigorr3) [reply data/membership-schema/src/main/resources/net/corda/schema/membership/mgm/registration/1.0/corda.mgm.registration.json]: I meant reformat it (I was being lazy and just copied the whole thing).

Comment (williamvigorr3) [reply data/membership-schema/src/main/resources/net/corda/schema/membership/mgm/registration/1.0/corda.mgm.registration.json]: What you did looks great.

**Review by williamvigorr3 — APPROVED** [review]
LGTM

**Review by charlieR3 — APPROVED** [review]

**Pull request merged**